starfish/src/main/java/org/maxandersen/Starfish.java

```diff
@@ -0,0 +1,238 @@
+    
+    for(String s:args)
+    System.out.println("\nTarget : "+s);//Printing Received Arguements for Debugging
+
+    if(args[0].equalsIgnoreCase("config")){
+    //Incase user wants to configure starfish
+    editConfig();//Calling function that enables to edit configuration
+    return 10; //Incase user typed "starfish config" we only want to edit configuration
+    }
+
+    //Calling function for Fetching installed Packages 
+    //fetch_installed_packages(); //This function will print all installed packages
+
+    //Configuration identifiers
+    String clone_path="";//Holds path to  destination Where the Repository Must Be Clonned
+    String ide="";  //Holds command for IDE to open upon
+    String userHome = System.getProperty( "user.home" ); //Get User Home Directory: /home/user_name
+
+    File configFile = new File(userHome + "/starfish-config.yml"); //Loading YAML
+
+
+  
+    
+    
+    //Reading Configuration File
+    try {
+        if (!configFile.exists()) {
+            //Incase no config file exists
+            System.out.println("\nThis is the first time you're using starfish...You'll need to  configure it");
+            editConfig(); //Calling function that lets user to configure 
+        }
+        
+        System.out.println("\nLoading configurations.....\n");
+        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
+    
+        Config config = mapper.readValue(configFile, Config.class);
+        
+        ide=config.ide;
+        clone_path=config.clone_path;
+        if(ide==null||clone_path==null)editConfig(); //Incase of absence of configuration in file Launch Config 
+        
+        
+    }
+    catch (Exception e) {
+        e.printStackTrace();
+        System.out.print("Press any key to continue . . . ");
+    }
+    
+
+
+    //Clonnning Git Repo
+    //Expected parameter: https://github.com/user-name/repo-name.git
+    String repo_name=args[0].substring(args[0].lastIndexOf("/"),args[0].lastIndexOf(".")); //Extracts the Name of Repository
+    //String clone_path= "/home/fahad/MyProjects/starfish_clonned/"; 
+    String cmd = "git clone "+args[0];
+    String originUrl = args[0];
+    Path directory = Paths.get(clone_path+repo_name);
+    gitClone(directory, originUrl);
+    
+
+
+
+    //Launching Vscode on the Cloned Directory 
+    System.out.println("Launching  IDE Now");
+    runCommand(directory.getParent(), ide,clone_path+repo_name);
+    return 10;
+    
+}
+
+
+//Function to edit configuration and serves for command line starfish config
+public static void editConfig()throws Exception{
+System.out.println("\n------Starfish Configuration Editor------");
+String clone_path="";//Holds path to  destination Where the Repository Must Be Clonned
+    String ide="";  //Holds command for IDE to open upon
+    String userHome = System.getProperty( "user.home" ); //Get User Home Directory: /home/user_name
+
+    File configFile = new File(userHome + "/starfish-config.yml"); //Loading YAML
```

Comment:
Second time with logic to look up config. Should be shared code.

Reply:
done